File: 0.1 Calculator.py
# Напишем простой калькулятор! Он будет читать 3 строки:

# первое число
# второй номер
# арифметическая операция.
# Числа - плавающие!

# Выходные данные являются результатом следующего: first_number операции second_number.

# Доступны следующие операции: +, -, /, *, mod, pow, div.
# mod - операция по модулю, т.е. остаток от деления first_number % second_number,
# pow - возведение в степень, первое число будет основанием, второе - степенью: first_number ** second_number,
# div - целочисленное деление first_number // second_number.

# Обратите внимание, что если второе число равно 0 и вы хотите выполнить любую из операций /, mod или div,
# калькулятор должен сказать «Division by 0!»

# Sample Input 1:
# 5.0
# 0.0
# mod
# Sample Output 1:
# Division by 0!

# Sample Input 2:
# -12.0
# -8.0
# *
# Sample Output 2:
# 96.0

# Sample Input 3:
# 5.0
# 10.0
# /
# Sample Output 3:
# 0.5

num1, num2, oper = float(input()), float(input()), input()

def main(a, b, oper):
    if oper == '+': return a + b
    elif oper == '-': return a - b
    elif oper == '/': return 'Division by 0!' if b == 0 else a / b
    elif oper == '*': return a * b
    elif oper == 'mod': return 'Division by 0!' if b == 0 else a % b
    elif oper == 'pow': return a ** b
    elif oper == 'div': return 'Division by 0!' if b == 0 else a // b
    else: return 'Wrong operator!'

print(main(num1, num2, oper))

File: 0.1 Calculator.c
#include <stdio.h>

double power(double x, int y){
    if (y == 0) return 1;
    return x * power(x, y-1);
}
int main (){
    char *mod = "mod";
    char *pow = "pow";
    char *div = "div";
    double x, y;
    scanf("%lf", &x);
    scanf("%lf", &y);
    char sign[3];
    scanf(" %s", sign);
    char f = sign[0];

    switch(f){
        case '+':
            printf("%.2lf", x+y);
            break;
        case '-':
            printf("%.2lf", x-y);
            break;
        case '/':
            if (y == 0){
                printf("Division by 0!\n");
                return 5;
            }
            printf("%.2lf", x/y);
            break;
        case '*':
            printf("%.2lf", x*y);
            break;
        case 'm':
            if (*sign != *mod) {
                printf("Wrong operator");
                return 6;
            }
            if (y == 0){
                printf("Division by 0!\n");
                return 5;
            }
            printf("%.2lf", x/y);
            break;
        case 'p':
            if (*sign != *pow) {
                printf("Wrong operator");
                return 6;
            }
            printf("%.2lf", power(x,(int)y));
            break;
        case 'd':
            if (*sign != *div) {
                printf("Wrong operator");
                return 6;
            }
            if (y == 0){
                printf("Division by 0!\n");
                return 5;
            }
            printf("%d", (int)x/(int)y);
            break;
        default:
            printf("Operation not allowed");
            return 1;
    }
    return 0;
}
